cpp/BUILD.bazel

```diff
@@ -157,6 +157,7 @@ cc_test(
         "counter.so",
         "plus.so",
         "ray_cpp_pkg",
+        "src/ray/test/cluster/ray-test-SNAPSHOT.jar",
```

Comment:
Can we move the xlang test to a separate test like `cluster_mode_xlang_test`? In some cases, we need to run the `cluster_mode_test ` in a simple environment which doesn't compile the java code or even doesn't install the JDK.

Reply:
cpp->python is different because c++ test depends on python anyway. I suggest move the java test first in current PR.

Reply:
Only x-lang tests depends on python.

Reply:
All the c++ tests depend on python because we should start ray cluster using `ray start --head`.